nativelib/src/main/resources/gc/immix/Heap.h

```diff
@@ -0,0 +1,47 @@
+#ifndef IMMIX_HEAP_H
+#define IMMIX_HEAP_H
+
+#include "GCTypes.h"
+#include "Allocator.h"
+#include "LargeAllocator.h"
+#include "datastructures/Stack.h"
+
+typedef struct {
+    word_t *heapStart;
+    word_t *heapEnd;
+    size_t smallHeapSize;
+    word_t *largeHeapStart;
+    word_t *largeHeapEnd;
+    word_t *largeHeapSize;
+    Allocator *allocator;
+    LargeAllocator *largeAllocator;
+} Heap;
+
+static inline bool heap_isWordInLargeHeap(Heap *heap, word_t *word) {
+    return word != NULL && word >= heap->largeHeapStart &&
+           word < heap->largeHeapEnd;
+}
+
+static inline bool heap_isWordInSmallHeap(Heap *heap, word_t *word) {
+    return word != NULL && word >= heap->heapStart && word < heap->heapEnd;
+}
+
+static inline bool heap_isWordInHeap(Heap *heap, word_t *word) {
+    return heap_isWordInSmallHeap(heap, word) ||
+           heap_isWordInLargeHeap(heap, word);
+}
```

Comment:
Can we make sure that large and small objects heaps are adjacent in memory? This way we'll need just two bounds checks here instead of 4.